[Android][API] Add GraphQL API doc for making queries with pagination (#1902)
aws-amplify/docs: docs/lib/graphqlapi/fragments/android/query-data.md

</amplify-block>
</amplify-block-switcher>

> **Note**: This approach will only return up to the first 1000 items.  To change this limit or make requests for additional results beyond this limit, see below.

## List multiple pages of items

> You can use this approach to specify the number of results per page (limit), and request a next page of results if available.

Reviewer: ```suggestion
Pagination allows you to request a maximum number of results to be returned in a response. To consume further results beyond that, you can request a next page of results, if available.
```

<amplify-block-switcher>
<amplify-block name="Java">

```java
public void queryFirstPage() {

>     query(ModelQuery.list(Todo.class, ModelPagination.firstPage().withLimit(1000)));

Reviewer: ```suggestion
    query(ModelQuery.list(Todo.class, ModelPagination.firstPage().withLimit(1_000)));
```

}

> public void query(GraphQLRequest<PaginatedResult<Todo>> request) {
>     Amplify.API.query(
>         request,
>         response -> {
>             if(response.hasData()) {
>                 for(Todo todo : response.getData().getItems()) {
>                     Log.d("MyAmplifyApp", todo.getName());
>                 }
>                 if(response.getData().hasNextResult()) {
>                     query(response.getData().getRequestForNextResult());
>                 }
>             }
>         },
>         error -> Log.e("MyAmplifyApp", "Query failure ", error)
>     );
> }

Reviewer: ```suggestion
private static void query(GraphQLRequest<PaginatedResult<Todo>> request) {
    Amplify.API.query(
        request,
        response -> {
            if (response.hasData()) {
                for (Todo todo : response.getData().getItems()) {
                    Log.d("MyAmplifyApp", todo.getName());
                }
                if (response.getData().hasNextResult()) {
                    query(response.getData().getRequestForNextResult());
                }
            }
        },
        failure -> Log.e("MyAmplifyApp", "Query failed.", failure)
    );
}
```

```

</amplify-block>
<amplify-block name="Kotlin">

```kotlin
fun queryFirstPage() {

>     query(ModelQuery.list(Todo::class.java, ModelPagination.firstPage().withLimit(1000)))

Reviewer: ```suggestion
    query(ModelQuery.list(Todo::class.java, ModelPagination.firstPage().withLimit(1_000)))
```

}

fun query(request: GraphQLRequest<PaginatedResult<Todo>>) {
    Amplify.API.query(
        request,
        { response ->
            if (response.hasData()) {
                for (todo in response.data.items) {
                    Log.d("MyAmplifyApp", todo.name)
                }
                if (response.data.hasNextResult()) {
                    query(response.data.requestForNextResult)
                }
            }
        },

>         { error -> Log.e("MyAmplifyApp", "Query failure ", error) }

Reviewer: ```suggestion
        { failure -> Log.e("MyAmplifyApp", "Query failed.", failure) }
```
